File: python/cms.py
import datetime, json, requests

def store_cms_summary_report():
    print("Getting CMS data")
    response = requests.get("https://jsonplaceholder.typicode.com/posts")
    if response.status_code == 200:
        try:
            posts = json.loads(response.text)
            print("CMS data has %s records" % len(posts))
        except Exception as e:
            print('No valid JSON response from CMS, error %s' % type(e).__name__)
        today = datetime.date.today().isoformat()
        report_filename = "cms-" + today + ".json"
        with open(report_filename, "w") as summary_file:
            summary = { "posts": 0, "users": 0, "mean_posts_per_user": 0 }
            users = []
            for post in posts:
                if post["userId"] not in users:
                    users.append(post["userId"])
            summary["posts"] = len(posts)
            summary["users"] = len(users)
            summary["mean_posts_per_user"] = round(summary["posts"] / summary["users"])
            summary_file.write(json.dumps(summary))
            print("Wrote CMS report", flush=True)
    else:
        print('Error getting data from CMS, response code %s' % response.status_code)


if __name__ == '__main__':
    store_cms_summary_report()

File: ruby/cms.rb
require 'date'
require 'net/http'

def generate_report
  today = Date.today.strftime("%F")
  puts "today's date is: #{today}"
  filename = "./cms-#{today}.json"
  puts "file is #{filename}"

  # Get posts from CMS
  puts 'Getting CMS data'
  url = 'http://jsonplaceholder.typicode.com/posts'
  uri = URI(url)
  response = Net::HTTP.get(uri)
  posts = JSON.parse(response)

  # Analyse data
  report = { 'posts' => 0, 'users' => 0, 'mean_posts_per_user' => 0 }
  report['posts'] = posts.length()
  users = []
  posts.each { |post|
    users.push(post['userId']) if !users.include?(post['userId'])
  }
  report['users'] = users.length()
  report['mean_posts_per_user'] = (posts.length() / users.length()).round
  puts "CMS data has #{posts.length()} posts"

  # Write report
  File.open(filename, 'w') { |file| file.write(report.to_json) }
  puts 'Wrote CMS report'
end
